R/produce_slf_ch_tests.R

```diff
@@ -0,0 +1,29 @@
+#' Care Home All Episodes Tests
+#'
+#' @description Produce the test for the Care Home
+#'
+#' @param data new or old data for testing summary flags
+#' (data is from [get_source_extract_path(year, "CH")])
+#'
+#' @return a dataframe with a count of each flag.
+#'
+#' @export
+#' @family social care test functions
+produce_slf_ch_tests <- function(data) {
+  data %>%
+    # create test flags
+    create_demog_test_flags() %>%
+    # remove variables that won't be summed
+    dplyr::select(
+      .data$valid_chi,
+      .data$unique_chi,
+      .data$n_missing_chi,
+      .data$n_males,
+      .data$n_females,
+      .data$n_postcode,
+      .data$n_missing_postcode,
+      .data$missing_dob
+    ) %>%
```

Comment:
I've checked the CH tests script and i think there are some flags you are missing here. Have a look at B08X Care home tests I think you are missing tests for: CH provider, SC support and sc_send_lca which some of these can be put into a function

Reply:
Hi, @Jennit07. I found the test variables from SPSS script include those as follows.  Do I need to include all in the R script? Apologize if my memory is wrong. I remember James mentioned those variables starting with "sc_" are not needed. Can you please confirm that? Are there any variables I am missing? Thank you!

-     /n_ch_name_missing 
-     /n_ch_provider_1_to_5
-     /n_ch_provider_other
-     /n_ch_adm_reason_missing
-     /n_ch_nursing
-     /n_sc_living_alone_no
-     /n_sc_living_alone_yes 
-     /n_sc_living_alone_unknown
-     /n_sc_support_from_unpaid_carer_no
-     /n_sc_support_from_unpaid_carer_yes
-     /n_sc_support_from_unpaid_carer_unknown
-     /n_sc_social_worker_no 
-     /n_sc_social_worker_yes 
-     /n_sc_social_worker_unknown 
-     /n_sc_meals_no 
-     /n_sc_meals_yes 
-     /n_sc_meals_unknown 
-     /n_sc_day_care_no
-     /n_sc_day_care_yes
-     /n_sc_day_care_unknown

Reply:
Hi @lizihao-anu yes, those variables listed are not needed. If you look at the SPSS script we want to test for beddays and costs you can use the function `calculate_measures` to help with this, see other tests such as acute, mental health and maternity to see how this is implemented. The script also checks how many episodes are in each health board where you can use the function `create_hb_test_flags`. Let me know if you need any help with this or if you would like me to talk through how the tests work as they can be complicated. Thanks